### ISSUE TITLE
[Bug] SingleTaskVariationalGP raises a warning when using input_transform

### ISSUE THREAD

**Issue by crasanders** [pytorch/botorch#1824]
# 🐛 Bug

Calling `SingleTaskVariationalGP.posterior` with an `input_transform` raises a warning, whereas the equivalent call with  `SingleTaskGP` does not. I'm not sure if `input_transform` works correctly with `SingleTaskVariationalGP` or if I can safely interpret the resulting posterior. It also seems a little odd to me that this would be a warning and not an exception.

## To reproduce

** Code snippet to reproduce **
```python
import torch
from botorch.models import SingleTaskGP, SingleTaskVariationalGP
from botorch.models.transforms import Normalize

X = torch.rand((20, 1))
y = torch.sin(X)

model = SingleTaskGP(train_X=X, train_Y=y, input_transform=Normalize(1))
post = model.posterior(X) # No warning

model = SingleTaskVariationalGP(train_X=X, train_Y=y, input_transform=Normalize(1))
post = model.posterior(X) # Warning
```

** Stack trace/error message **
```
RuntimeWarning: Could not update `train_inputs` with transformed inputs since SingleTaskVariationalGP does not have a `train_inputs` attribute. Make sure that the `input_transform` is applied to both the train inputs and test inputs.```
```

## System information

botorch version = 0.8.5
gpytorch version = 1.10
torch version = 1.13.1

**Comment by Balandat** [comment]
cc @saitcakmak this is relevant for the proposed transforms refactor in https://github.com/cornellius-gp/gpytorch/pull/2114

**Comment by crasanders** [comment]
Great, glad it's a false alarm. Thanks for the quick reply.

**Comment by saitcakmak** [comment]
I vaguely recall some issue with inducing points sometimes getting transformed and sometimes not with ApproximateGP. Looking at my old notes, I found this:
> It does not play well with ApproximateGP, since inducing points sometimes get transformed (in train mode) and sometimes don’t (in posterior). This leads to bugs as we train them under one setup and evaluate them under another.

So, there might be some truth to the warning here. The proper solution would be to push through https://github.com/pytorch/botorch/pull/1372 and fix this for good. I'll leave this open just in case.
